drivers/timer/cortex_m_systick.c

```diff
@@ -103,9 +103,7 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	cycle_count += elapsed();
 
 	/* Round delay up to next tick boundary */
-	delay = delay + (cycle_count - announced_cycles);
-	delay = ((delay + CYC_PER_TICK - 1) / CYC_PER_TICK) * CYC_PER_TICK;
-	last_load = delay - (cycle_count - announced_cycles);
+	last_load = ((delay + CYC_PER_TICK - 1) / CYC_PER_TICK) * CYC_PER_TICK;
```

Comment:
This isn't quite right.  Tick boundaries are absolute times, this rounds up to an integer number of ticks from whatever the current time is when z_clock_set_timeout() is called.

The bug here (if I'm understanding the problem correctly) is that we're subtracting announced_cycles and not "time_of_the_previous_tick".

Reply:
I'm not sure to understand the purpose of the original code. We are adding (cycle_count - announced_cycles) to delay just for the sake of rounding, and then we are subtracting it soon after.

I saw that in many cases last_load would so result in a value slightly less than CYC_PER_TICK, which I think is wrong. In fact, I'm able to see a (I guess) correct behaviour of k_sleep keeping the original code but enforcing last_load to be at least CYC_PER_TICK:

 if (last_load < CYC_PER_TICK) 
     last_load = CYC_PER_TICK;

I would like to fix it for 1.14, but not sure how to do that properly.... Any idea?